### ISSUE TITLE
flux bootstrap - failed to clone repository: object not found (both git ssh:// & github token variations)

### ISSUE THREAD

**Issue by HariSekhon** [fluxcd/flux2#1463]
### Describe the bug

Flux bootstreap fails to clone GitHub repo while bootstrapping, whether using plain git with ssh:// or github with token and has insufficient debug logging.

SSH key and GitHub token have both been verified and used in other tools a lot.

An original bootstap for this cluster did work using `flux bootstrap ssh ...` and the generated SSH key was added to the repo's deploy keys.

Trying to re-setup the cluster has been failing like this and it's not clear if this is because the yamls already exist in the cluster (although iirc reading the docs this should be fine as it's how one would customize the install)

### To Reproduce

```
flux bootstrap git --url=ssh://git@github.com/MyOrg/MyRepo.git --branch=UN-16443-fluxcd --path=k8s/dev --log-level=debug
► cloning branch "UN-16443-fluxcd" from Git repository "ssh://git@github.com/MyOrg/MyRepo.git"
✗ failed to clone repository: object not found
```
Tried this with and without `.git` appended to the url, and have verified the `k8s/dev` path is available on the upstream branch `UN-16443-fluxcd` which contains the previous bootstrap yamls under `k8s/dev/flux-system`

### Expected behavior

Should reuse the yamls in the git repo, bootstrap an SSH key and prompt to add it to the deploy keys of the repo.

I have run this once before and have the bootstrap yamls committed in the upstream branch, as seen from the repo root checkout on the branch `UN-16443-fluxcd`:
```
$ ls -l k8s/dev/flux-system/
total 288
-rw-r--r--  1 harisekhon  staff  136601 May 28 18:08 gotk-components.yaml
-rw-r--r--  1 harisekhon  staff     529 May 28 18:08 gotk-sync.yaml
-rw-r--r--  1 harisekhon  staff     115 May 28 18:08 kustomization.yaml
```

I did even generate a flux Kustomization and GitRepository yamls and committed them to `k8s/dev/` directory.

### Additional context

Flux components are installed to namespace, all checks pass:

```
flux --version
flux version 0.14.1
```

```
$ flux check
► checking prerequisites
✔ kubectl 1.19.7 >=1.18.0-0
✔ Kubernetes 1.17.17-gke.4900 >=1.16.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.10.1
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.12.1
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.14.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.13.1
✔ all checks passed
```

```
$ kubectl -n flux-system get all
NAME                                           READY   STATUS    RESTARTS   AGE
pod/helm-controller-78b556b8d6-fgcnq           1/1     Running   0          96m
pod/kustomize-controller-8d7d96647-xzk29       1/1     Running   0          96m
pod/notification-controller-7bd669bc6d-xzqxp   1/1     Running   0          96m
pod/source-controller-b5b9dcc56-n47vp          1/1     Running   0          96m

NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
service/notification-controller   ClusterIP   10.128.2.249    <none>        80/TCP    96m
service/source-controller         ClusterIP   10.128.3.195    <none>        80/TCP    96m
service/webhook-receiver          ClusterIP   10.128.12.211   <none>        80/TCP    96m

NAME                                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/helm-controller           1/1     1            1           96m
deployment.apps/kustomize-controller      1/1     1            1           96m
deployment.apps/notification-controller   1/1     1            1           96m
deployment.apps/source-controller         1/1     1            1           96m

NAME                                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/helm-controller-78b556b8d6           1         1         1       96m
replicaset.apps/kustomize-controller-8d7d96647       1         1         1       96m
replicaset.apps/notification-controller-7bd669bc6d   1         1         1       96m
replicaset.apps/source-controller-b5b9dcc56          1         1         1       96m
```

```
kubectl -n flux-system logs deploy/source-controller
{"level":"info","ts":"2021-06-01T10:20:42.946Z","logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":"2021-06-01T10:20:42.959Z","logger":"setup","msg":"starting manager"}
{"level":"info","ts":"2021-06-01T10:20:42.964Z","msg":"starting metrics server","path":"/metrics"}
I0601 10:20:42.966702       6 leaderelection.go:243] attempting to acquire leader lease flux-system/source-controller-leader-election...
I0601 10:20:43.096139       6 leaderelection.go:253] successfully acquired lease flux-system/source-controller-leader-election
{"level":"info","ts":"2021-06-01T10:20:43.269Z","logger":"setup","msg":"starting file server"}
{"level":"info","ts":"2021-06-01T10:20:43.269Z","logger":"controller.gitrepository","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.helmrepository","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmRepository","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.helmrepository","msg":"Starting Controller","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmRepository"}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.helmchart","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.helmchart","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.helmchart","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.270Z","logger":"controller.bucket","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"Bucket","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.bucket","msg":"Starting Controller","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"Bucket"}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.bucket","msg":"Starting workers","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"Bucket","worker count":2}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.gitrepository","msg":"Starting Controller","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository"}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.gitrepository","msg":"Starting workers","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","worker count":2}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.helmrepository","msg":"Starting workers","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmRepository","worker count":2}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.helmchart","msg":"Starting EventSource","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.helmchart","msg":"Starting Controller","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart"}
{"level":"info","ts":"2021-06-01T10:20:43.373Z","logger":"controller.helmchart","msg":"Starting workers","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmChart","worker count":2}
```

```
kubectl -n flux-system logs deploy/kustomize-controller
{"level":"info","ts":"2021-06-01T10:20:52.389Z","logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":"2021-06-01T10:20:52.391Z","logger":"setup","msg":"starting manager"}
I0601 10:20:52.393167       6 leaderelection.go:243] attempting to acquire leader lease flux-system/kustomize-controller-leader-election...
{"level":"info","ts":"2021-06-01T10:20:52.393Z","msg":"starting metrics server","path":"/metrics"}
I0601 10:20:52.421551       6 leaderelection.go:253] successfully acquired lease flux-system/kustomize-controller-leader-election
{"level":"info","ts":"2021-06-01T10:20:52.495Z","logger":"controller.kustomization","msg":"Starting EventSource","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:52.495Z","logger":"controller.kustomization","msg":"Starting EventSource","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:52.596Z","logger":"controller.kustomization","msg":"Starting EventSource","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","source":"kind source: /, Kind="}
{"level":"info","ts":"2021-06-01T10:20:52.697Z","logger":"controller.kustomization","msg":"Starting Controller","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization"}
{"level":"info","ts":"2021-06-01T10:20:52.697Z","logger":"controller.kustomization","msg":"Starting workers","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","worker count":4}
```

**Comment by HariSekhon** [comment]
After removing the `flux-system` namespace a couple of times and failing to bootstrap as shown above using both `git` with ssh:// and `github` with https:// URLs I ended up doing this to get up and running and reconciling successfully:

```
flux install
```

```
flux create source git...
```
(had to run the command and not just apply my previously saved fluxcd-repo.yaml to get a new SSH key pair to load to GitHub's deploy keys)
```
kubectl apply -f fluxcd-kustomization.yaml
```

**Comment by tricky42** [comment]
Just today I ran into the same problem. Suddenly running 

`flux bootstrap github --context=my-context --owner=my-corp --repository=cloud --branch=feat-crossplane --path=stacks/dev-abu/mcp` fails with:

```
+--------------------------------------------------------------+
| Initiate Flux Bootstrap Process...                           |
+--------------------------------------------------------------+
► connecting to github.com
► cloning branch "feat-crossplane" from Git repository "https://github.com/my-corp/cloud.git"
✗ failed to clone repository: object not found
```

Cloning the repository in the shell using `git`-CLI works without a problem: 
```
git clone https://github.com/my-corp/cloud.git .
Cloning into '.'...
remote: Enumerating objects: 30173, done.
remote: Counting objects: 100% (5242/5242), done.
remote: Compressing objects: 100% (2637/2637), done.
remote: Total 30173 (delta 2568), reused 4852 (delta 2383), pack-reused 24931
Receiving objects: 100% (30173/30173), 151.86 MiB | 22.36 MiB/s, done.
Resolving deltas: 100% (16704/16704), done.
```

Using option `--verbose` doesn't provide more details :(

**Comment by stefanprodan** [comment]
Please use flux2 v0.13.4, we've updated go-git to v5.4.0 and that broke Flux for repos with large files, upstream bug: https://github.com/go-git/go-git/issues/323

**Comment by stefanprodan** [comment]
Fixed in flux2 0.14.2
